src/H5Ocache.c

```diff
@@ -1127,14 +1022,20 @@ H5O__prefix_deserialize(const uint8_t *_image, H5O_cache_ud_t *udata)
     /* (indicates version 2 or later) */
     if (!HDmemcmp(image, H5O_HDR_MAGIC, (size_t)H5_SIZEOF_MAGIC)) {
         /* Magic number */
+        if (H5_IS_BUFFER_OVERFLOW(image, H5_SIZEOF_MAGIC, p_end))
```

Comment:
Should probably move this check outside the block before the memcmp call

Reply:
Fixed